src/gei/Handlers/MigrateRepoCommandHandler.cs

```diff
@@ -236,13 +238,17 @@ private async Task<string> WaitForArchiveGeneration(GithubApi githubApi, string
         {
             var archiveStatus = await githubApi.GetArchiveMigrationStatus(githubSourceOrg, archiveId);
             _log.LogInformation($"Waiting for archive with id {archiveId} generation to finish. Current status: {archiveStatus}");
-            if (archiveStatus == ArchiveMigrationStatus.Exported)
+            if (archiveStatus == ArchiveMigrationStatus.Failed)
             {
-                return await githubApi.GetArchiveMigrationUrl(githubSourceOrg, archiveId);
+                var result = await _retryPolicy.RetryOnResult(async () => await githubApi.GetArchiveMigrationStatus(githubSourceOrg, archiveId), ArchiveMigrationStatus.Failed);
```

Comment:
this probably applies to all `GetArchiveMigrationStatus` calls, so it'd be better to move this retry logic into the `GithubApi` itself and eventually return the migration status. It would also be better separation of concerns.

Reply:
I moved the retry logic into GithubApi here: 01a9f99107172c274bf39daf8c6ea6fa447e76cd